src/pytorch_lightning/utilities/apply_func.py

```diff
@@ -342,7 +347,8 @@ def batch_to(data: Any) -> Any:
 
         kwargs = {}
         # Don't issue non-blocking transfers to CPU
-        if isinstance(data, Tensor) and device not in _CPU_DEVICES:
+        # Same with MPS due to a race condition bug: https://github.com/pytorch/pytorch/issues/83015
+        if isinstance(data, Tensor) and device not in _CPU_DEVICES and device not in _MPS_DEVICES:
```

Comment:
```suggestion
        if isinstance(device, str):
             device = torch.device(device)
             
        if isinstance(data, Tensor) and device.type not in _BLOCKING_DEVICES:
```

Reply:
Oh, I quite like this solution. It is a bit more elegant. As a side benefit, we then also do **not** need to make these import changes as proposed by @carmocca. So should I undo these import changes, or should I leave them in this PR?